src/main/java/com/google/api/codegen/transformer/TestCaseTransformer.java

```diff
@@ -233,8 +236,9 @@ public TestCaseView createTestCaseView(
         .build();
   }
 
-  private String getMethodDescriptorName(MethodContext context) {
-    if (context.getProductConfig().getTransportProtocol().equals(TransportProtocol.HTTP)) {
+  private String getMethodDescriptorName(MethodContext context, TargetLanguage targetLanguage) {
+    if (context.getProductConfig().getTransportProtocol().equals(TransportProtocol.HTTP)
+        && targetLanguage != TargetLanguage.PHP) {
```

Comment:
This is an unfortunate hack to get around the issue with Java doing something different than PHP. If there's a better way to do this, I'm all ears!

Reply:
For this particular method, probably just make it Java-only (instead everything but PHP). I checked why this method even exists, and it turns out it should not even be there (it is  java-specific logic which should have been refactored long time ago).